test/dotnet-watch.FunctionalTests/ProjectToolScenario.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Watcher.FunctionalTests
+{
+    public class ProjectToolScenario: IDisposable
+    {
+        private const string NugetConfigFileName = "NuGet.Config";
+
+        public ProjectToolScenario()
+        {
+            Console.WriteLine($"The temporary test folder is {TempFolder}");
+
+            WorkFolder = Path.Combine(TempFolder, "work");
+            PackagesFolder = Path.Combine(TempFolder, "packages");
+
+            CreateTestDirectory();
+        }
+
+        public string TempFolder { get; } = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+
+        public string WorkFolder { get; }
+
+        public string PackagesFolder { get; }
+
+        public void AddProject(string projectFolder)
+        {
+            var destinationFolder = Path.Combine(WorkFolder, Path.GetFileName(projectFolder));
+            Console.WriteLine($"Copying project {projectFolder} to {destinationFolder}");
+
+            Directory.CreateDirectory(destinationFolder);
+
+            foreach (string directory in Directory.GetDirectories(projectFolder, "*", SearchOption.AllDirectories))
+            {
+                Directory.CreateDirectory(directory.Replace(projectFolder, destinationFolder));
+            }
+
+            foreach (string file in Directory.GetFiles(projectFolder, "*.*", SearchOption.AllDirectories))
+            {
+                File.Copy(file, file.Replace(projectFolder, destinationFolder), true);
+            }
+        }
+
+        public void AddNugetFeed(string feedName, string feed)
+        {
+            string tempNugetConfigFile = Path.Combine(WorkFolder, NugetConfigFileName);
+
+            var nugetConfig = XDocument.Load(tempNugetConfigFile);
+            var packageSource = nugetConfig.Element("configuration").Element("packageSources");
+            packageSource.Add(new XElement("add", new XAttribute("key", feedName), new XAttribute("value", feed)));
+            using (var stream = File.OpenWrite(tempNugetConfigFile))
+            {
+                nugetConfig.Save(stream);
+            }
+        }
+
+        public void AddToolToProject(string projectName, string toolName)
+        {
+            var projectFile = Path.Combine(WorkFolder, projectName, "project.json");
+            Console.WriteLine($"Adding {toolName} to {projectFile}");
+
+            var projectJson = JObject.Parse(File.ReadAllText(projectFile));
+            projectJson.Add("tools", new JObject(new JProperty(toolName, "1.0.0-*")));
```

Comment:
Will this update the tools section if one exists or create a new "tools" section if it does not exist? Also I think in general you should take the version as a parameter (even if it had the default value of 1.0.0-*)